controllers/user_controller.go

```diff
@@ -0,0 +1,428 @@
+		if err != nil {
+			logger.Error(err, "An error occurred while adding groups to user", "user", userResource.Spec.Name)
+			return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "S3UserCreationFailed",
+				fmt.Sprintf("Error while updating groups of user [%s] on S3 instance has failed", userResource.Spec.Name), err)
+		}
+	}
+
+	// Define a new K8S Secrets
+	data := make(map[string][]byte)
+	data["accessKey"] = []byte(username)
+	data["secretKey"] = []byte(password)
+	secret, err := newSecretForCR(userResource, data)
+
+	if err != nil {
+		// Error while creating the Kubernetes secret - requeue the request.
+		logger.Error(err, "Could not generate Kubernetes secret")
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretGenerationFailed",
+			fmt.Sprintf("Generation of k8s secrets [%s] has failed", userResource.Spec.Name), err)
+	}
+
+	// Set S3User instance as the owner and controller
+	err = ctrl.SetControllerReference(userResource, secret, r.Scheme)
+	if err != nil {
+		logger.Error(err, "Could not set owner of kubernetes secret")
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretGenerationFailed",
+			fmt.Sprintf("Generation of k8s secrets [%s] has failed", userResource.Spec.Name), err)
+	}
+
+	// Check if this Secret already exists
+	err = r.Get(ctx, types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}, &corev1.Secret{})
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Creating a new Secret", "Secret.Namespace", secret.Namespace, "Secret.Name", secret.Name)
+		err = r.Create(ctx, secret)
+		if err != nil {
+			logger.Error(err, "Could not create secret")
+			return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretCreationFailed",
+				fmt.Sprintf("Generation of k8s secrets [%s] has failed", secret.Name), err)
+		}
+		// Secret created successfully - requeue only if no version is specified
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorSucceeded", metav1.ConditionTrue, "SecretCreationSuccess",
+			fmt.Sprintf("The secret [%s] was created according to its matching custom resource", secret.Name), nil)
+	} else if err != nil {
+		logger.Error(err, "Could not create secret")
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretCreationFailed",
+			fmt.Sprintf("Creation of k8s secrets [%s] has failed", secret.Name), err)
+	} else {
+		if r.OverrideExistingSecret {
+			logger.Info("A secret with the same name already exist you choose to override existing secret. Deleting secret", "Secret.Namespace", secret.Namespace, "Secret.Name", secret.Name)
+			logger.Info("Updating the Secret", "Secret.Namespace", secret.Namespace, "Secret.Name", secret.Name)
+			err = r.Update(ctx, secret)
```

Comment:
Might have missed something, but there appears to be no actual secret deletion, just an update ?

Reply:
Yep, it was a bad log comment, i change it :)